src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/ProjectSystem/DefaultProjectResolver.cs

```diff
@@ -59,22 +59,32 @@ public override bool TryResolvePotentialProject(string documentFilePath, out Pro
             var projects = _projectSnapshotManagerAccessor.Instance.Projects;
             for (var i = 0; i < projects.Count; i++)
             {
-                if (projects[i].FilePath == _miscellaneousHostProject.FilePath)
+                projectSnapshot = projects[i];
+
+                if (projectSnapshot.FilePath == _miscellaneousHostProject.FilePath)
                 {
+                    if (IsDocumentInProject(projectSnapshot, documentFilePath))
+                    {
+                        return true;
+                    }
+
                     // We don't resolve documents to belonging to the miscellaneous project.
                     continue;
                 }
 
-                var projectDirectory = _filePathNormalizer.GetDirectory(projects[i].FilePath);
-                if (normalizedDocumentPath.StartsWith(projectDirectory, FilePathComparison.Instance))
+                var projectDirectory = _filePathNormalizer.GetDirectory(projectSnapshot.FilePath);
+                if (normalizedDocumentPath.StartsWith(projectDirectory, FilePathComparison.Instance) &&
+                    IsDocumentInProject(projectSnapshot, documentFilePath))
```

Comment:
Interesting, this got me thinking. Prior to this when you first open a document (without a corresponding project.razor.json) we'd resolve the project we think it should belong to. Now we will fail to add a document that doesn't have a corresponding project.razor.json entry. 

That being said, if we were to remove this "IsDocumentInProject" check then if a file was already in the MISC project we wouldn't remove it from the MISC project until a project.razor.json was added for it so it'd also defeat the purpose of trying to resolve the project it will belong to...

Maybe this actually isn't the right fix and we should be updating all the call sites to understand that the "potential project" may not actually have the document in it.

Reply:
Aside from the [migrate commands](https://github.com/dotnet/aspnetcore-tooling/blob/master/src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/ProjectSystem/DefaultRazorProjectService.cs#L411-L465) is there any place we'd want to resolve a potential project which doesn't contain the actual file?

I'd assume for all the other stuff:

1. TryResolveDocument
2. OpenDocument
3. AddDocument
4. UpdateDocument
5. CloseDocument
6. RemoveDocument

we'd want the project where the file actually is.

All uses of `TryResolvePotentialProject` are limited to the [`DefaultRazorProjectService`](https://github.com/dotnet/aspnetcore-tooling/blob/master/src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/ProjectSystem/DefaultRazorProjectService.cs)

Reply:
AddDocument we'd want the "potential" project if it didn't exist already